components/layout/display_list_builder.rs

```diff
@@ -2053,22 +2039,20 @@ impl FragmentDisplayListBuilding for Fragment {
                                             state: &mut DisplayListBuildState,
                                             text_fragment: &ScannedTextFragmentInfo,
                                             stacking_relative_content_box: &Rect<Au>,
-                                            text_shadow: Option<&SimpleShadow>,
+                                            text_shadows: &[SimpleShadow],
                                             clip: &Rect<Au>) {
+
+        // NB: text painting order (CSS Text Decoration Module Level 3):
+        // shadows, underline, overline, text, text-emphasis, line-through
```

Comment:
This comment is a little bit unclear to me. Perhaps it could be something like:

```
// NB: The order for painting text components (CSS Text Decoration Module Level 3) is shadows
// underline, overline, text, text-emphasis, and then line-through.
```

I think making it a complete sentence with the verb and such makes it a little clearer.